### PR TITLE
Complex

### PR DESCRIPTION
**Pull request by rdaly525**
Adds a library of complex operations and example test. This sets the way for the complex floating point ops along with things like 32 bit add.

### CODE REVIEW

**Comment by Kuree** [comment]
@rdaly525 Would you please make sure things are working locally or in your branch before you fire an PR? You are using all the available travis resources and queued up tons of jobs. Just want to be considerate to others.

image

**Comment by rdaly525** [comment]
@Kuree, sorry I will try to use less resources. Most of the time I am trying to fix a travis error that I cannot reproduce in my local branch. Do you have a good testing flow for travis scripts?

**Comment by leonardt** [comment]
Enabling auto-cancellation could help limit builds for a single branch (only one running for the latest commit), see https://blog.travis-ci.com/2017-09-21-default-auto-cancellation

**Comment by leonardt** [comment]
Here's a pointer to the doc page: https://docs.travis-ci.com/user/customizing-the-build/#building-only-the-latest-commit

**Comment by Kuree** [comment]
I'm in the process of moving every tests into docker containers that can reproduced everywhere. That has been done for `garnet`, `canal`, and `gemstone` and it's been working very well so far. I will file a PR once it's done and you should be able to test locally easily.

**Comment by rdaly525** [comment]
Looks like auto cancellation is already enabled in lassen from travis (although I think I have seen otherwise...)

**Comment by rdaly525** [comment]
@cdonovick travis is passing, can you approve?

File: tests/test_complex.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from hwtypes import BitVector, Bit
+from lassen.sim import gen_pe
+import lassen.asm as asm
+
+Bit = Bit
+Data = BitVector[DATAWIDTH]
+
+FMA = gen_FMA(BitVector.get_family())
+
+def test_fma():
```
Comment (cdonovick) [inline tests/test_complex.py]: Shoud do random testing as done in test_pe not just a single set of valyes

Comment (rdaly525) [reply tests/test_complex.py]: Agreed, but let me do that in a different PR.

Comment (rdaly525) [reply tests/test_complex.py]: This is more a simple infrastructure PR

Comment (cdonovick) [reply tests/test_complex.py]: Does it not work with randomized inputs? If so I probably shouldn't be merged else new code should come with testing.

Comment (rdaly525) [reply tests/test_complex.py]: Added more parameterized tests.

**Pull request merged**